File: Insertar valor/valor.cpp
#include <iostream>
#include <vector>

std::vector<int> insertarEnIndice(std::vector<int> lista, int valor, int indice) {
    lista.insert(lista.begin() + indice, valor);
    return lista;
}

int main() {
    std::vector<int> miLista = {1, 2, 3, 5};
    miLista = insertarEnIndice(miLista, 4, 3);
    for (int num : miLista) std::cout << num << " ";
    return 0;
}

File: Insertar valor/valor.cs
using System;
using System.Collections.Generic;

class Program {
    static List<int> InsertarEnIndice(List<int> lista, int valor, int indice) {
        lista.Insert(indice, valor);
        return lista;
    }

    static void Main() {
        List<int> miLista = new List<int> {1, 2, 3, 5};
        miLista = InsertarEnIndice(miLista, 4, 3);
        Console.WriteLine(string.Join(", ", miLista));
    }
}